### ISSUE TITLE
Cassandra support broken: NoClassDefFoundError: io/netty/util/Timer

### ISSUE THREAD

**Issue by snicoll** [spring-projects/spring-boot#6616]
The cassandra support is broken: running the samples leads to 

```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.datastax.driver.core.Cluster]: Factory method 'cluster' threw exception; nested exception is java.lang.NoClassDefFoundError: io/netty/util/Timer
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588)
	... 71 more
Caused by: java.lang.NoClassDefFoundError: io/netty/util/Timer
	at com.datastax.driver.core.Cluster$Builder.<init>(Cluster.java:685)
	at com.datastax.driver.core.Cluster.builder(Cluster.java:193)
	at org.springframework.boot.autoconfigure.cassandra.CassandraAutoConfiguration.cluster(CassandraAutoConfiguration.java:57)
	at org.springframework.boot.autoconfigure.cassandra.CassandraAutoConfiguration$$EnhancerBySpringCGLIB$$ddf6c6a7.CGLIB$cluster$0(<generated>)
	at org.springframework.boot.autoconfigure.cassandra.CassandraAutoConfiguration$$EnhancerBySpringCGLIB$$ddf6c6a7$$FastClassBySpringCGLIB$$c64753a1.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:356)
	at org.springframework.boot.autoconfigure.cassandra.CassandraAutoConfiguration$$EnhancerBySpringCGLIB$$ddf6c6a7.cluster(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	... 72 more
Caused by: java.lang.ClassNotFoundException: io.netty.util.Timer
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 85 more
```

The unit tests pass however because the test scope has `cassandra-unit` which brings the necessary dependency.

**Comment by lnalex** [comment]
I have the same issue on a project, adding  `io.netty:netty-all` dependency solves it but waiting for a fix asap.

**Comment by wilkinsona** [comment]
Do we have a Spring Data Cassandra issue open for this? It's the dependency management in its pom excluding `netty-all` that's the root cause.

/cc @mp911de

**Comment by mp911de** [comment]
Spring Data Cassandra 1.4 ships without the netty dependency. That change (exclusion of netty) happened with Spring Data 1.4 (Hopper). Spring Data 1.3 (Gosling) included the netty dependency.

**Comment by wilkinsona** [comment]
@mp911de Thanks. That matches this breaking in Boot 1.4 which uses Hopper. What's the rationale for preventing the Driver from pulling in its Netty dependency? It seems to be required.

**Comment by mp911de** [comment]
The former developers of Spring Data Cassandra excluded netty because of build-time version conflicts with the embedded Cassandra database that was/is used during the test. I created [DATACASS-331](https://jira.spring.io/browse/DATACASS-331) to fix the netty dependency issue.

**Comment by wilkinsona** [comment]
@mp911de Thanks.

We're already building against Hopper snapshots so we'll hopefull pick up any fix in due course. However, I'll keep this issue open for now just in case nothing can be done on the Spring Data side.

**Comment by mp911de** [comment]
The issue is fixed in Spring Data Cassandra 1.4.3 (Hopper SR3 Snapshot) and 1.5.0 (Ingalls RC1 Snapshot). `netty-handler` is excluded as dependency from `cassandra-driver-dse` in [spring-boot-dependencies/pom.xml](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-dependencies/pom.xml#L600). I created #6721 to include `netty-handler` again.
